internal/pkg/util/fs/overlay/overlay_set_linux.go

```diff
@@ -80,9 +103,36 @@ func (s Set) performIndividualMounts() error {
 func (s Set) performFinalMount(rootFsDir string) error {
 	// Try to perform actual mount
 	options := s.options(rootFsDir)
-	sylog.Debugf("Mounting overlay with rootFsDir %q, options: %q", rootFsDir, options)
-	if err := syscall.Mount("overlay", rootFsDir, "overlay", syscall.MS_NODEV, options); err != nil {
-		return fmt.Errorf("failed to mount %s: %s", rootFsDir, err)
+	unprivOls, err := UnprivOverlaysSupported()
+	if err != nil {
```

Comment:
```suggestion
	if err != ErrNoRootlessOverlay {
```

Reply:
Thanks for the catch! (And the work that no doubt went into setting up a RHEL7 environment in which to do so.) But with the suggested change, if the error _is_ nil (as in systems with unprivileged overlay support), it'll go into the error block, right? I think it would be better to change UnprivOverlaysSupported() so that it returned nil in the error field under the relevant circumstances.